### PR TITLE
Creation of the How-To Contribute to docs

### PR DESCRIPTION
**Pull request by antoine2711**
In the structure of the new documentation, a section should be added that would contain How-Tos, pages of instructions on a particular subject/task.

The first How-To I'm creating is a « How-to Contribute to OpenRefine’s documentation ».

I'm looking for feedback and suggestions here.

Regards,
   Antoine

### CODE REVIEW

File: docs/src/how-tos/build_from_source.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+---
+
+ 
+You will need:
+* [OpenRefine source code](https://github.com/OpenRefine/OpenRefine)
+* [Java JDK](http://java.sun.com/javase/downloads/index.jsp)
+* [Apache Maven](https://maven.apache.org)  (OPTIONAL)
+* A Unix/Linux shell environment OR the Windows command line
+
+From the top level directory in the OpenRefine application you can build, test and run OpenRefine using the `./refine` shell script (if you are working in a *nix shell), or using the `refine.bat` script from the Windows command line. Note that the `refine.bat` on Windows only supports a subset of the functionality, supported by the `refine` shell script. The example commands below are using the `./refine` shell script, and you will need to use `refine.bat` if you are working from the Windows command line.
```
Comment (wetneb) [inline docs/src/how-tos/build_from_source.md:14]: The `*` in `*nix` is opening an italicized section here (at least according to GitHub's markdown highlighting), so it should be escaped.

File: docs/src/how-tos/contribute_to_or.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+First, you have to download the OpenRefine project on your computer. To do so, you can use `git` or GitDesktop. You can also use a git client integrated in an IDE like Eclipse.
+
+Then, you can configure [Docusaurus](https://docusaurus.io/), the tool the OpenRefine uses to manage it's documentation library.
+
+```sh
+cd docs
+yarn; yarn build; yarn start
+```
+
+This process is explained in more details in [Docusaurus Installation](/Docusaurus_Install.md).
```
Comment (kushthedude) [inline docs/src/how-tos/contribute_to_or.md:21]:
```suggestion
This process is explained in more details at [Docusaurus Installation](/Docusaurus_Install.md).
```

File: docs/src/how-tos/contribute_to_or.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+Once you have Docusaurus setup on your computer (you can install and run the Docusaurus server, and be able to see your modification as you go), you are ready to make you changes.
```
Comment (kushthedude) [inline docs/src/how-tos/contribute_to_or.md:23]:
```suggestion
Once you have Docusaurus setup on your computer (you can install and run the Docusaurus server, and be able to see the modification as you make), you are ready to make your changes.
```

File: docs/src/how-tos/contribute_to_or.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+To keep track of your work, create a pull request on GitHub OpenRefine/OpenRefine repertory, and commit your changes there. Once you have achieve a substantial amount of work, you can now ask that your changes be reviewed. At this stage, members of the community may suggest changes to your work, in order to be better integrated with the overall of OpenRefine's documentation.
```
Comment (kushthedude) [inline docs/src/how-tos/contribute_to_or.md:25]:
```suggestion
To keep track of your work, create a pull request on GitHub OpenRefine/OpenRefine repository, and commit your changes there. Once you have achieved a substantial amount of work, you can now ask that your changes be reviewed. At this stage, members of the community may suggest changes to your work, in order to be better integrated with the overall of OpenRefine's documentation.
```

File: docs/src/how-tos/contribute_to_or.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+Note that OpenRefine's documentation shares the same repertory as the OpenRefine source code base. So, your PR for the documentation changes will share the same space as the code modification PRs. You might want to add the filter « documentation » to GitHub searches, like this : « `is:open label:documentation` ».
```
Comment (kushthedude) [inline docs/src/how-tos/contribute_to_or.md:27]:
```suggestion
Note that OpenRefine's documentation shares the same repository as the OpenRefine source codebase. So, your PR for the documentation changes will share the same space as the code modification PRs. You might want to add the filter « documentation » to GitHub searches, like this : « `is:open label:documentation` ».
```

File: docs/src/how-tos/contribute_to_or.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+---
+
+
+If you are interested to help make OpenRefine’s documentation better, here's some informations.
+
+The documentation is separated in the Product Reference part, and How-to. There is some documentation mainly written for Users, and there is some documentation more geared toward the development of OpenRefine. 
+
+First, you have to download the OpenRefine project on your computer. To do so, you can use `git` or GitDesktop. You can also use a git client integrated in an IDE like Eclipse.
+
+Then, you can configure [Docusaurus](https://docusaurus.io/), the tool the OpenRefine uses to manage it's documentation library.
```
Comment (kushthedude) [inline docs/src/how-tos/contribute_to_or.md:14]:
```suggestion
Then, you can configure [Docusaurus](https://docusaurus.io/), the tool which OpenRefine uses to manage its documentation library.
```

File: docs/src/how-tos/contribute_to_or.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
+---
+id: howto-contribute-or
+title: How-To Contribute to OpenRefine's documentation
+sidebar_label: How-To Contribute to docs
+---
+
+
+If you are interested to help make OpenRefine’s documentation better, here's some informations.
+
+The documentation is separated in the Product Reference part, and How-to. There is some documentation mainly written for Users, and there is some documentation more geared toward the development of OpenRefine.
```
Comment (kushthedude) [inline docs/src/how-tos/contribute_to_or.md:10]:
```suggestion
The documentation is separated in the Product Reference part, and How-to. There is some documentation mainly written for Users, and there is some documentation more geared towards the development of OpenRefine. 
```

File: docs/src/how-tos/contribute_to_or.md (hunk 6 above)
Comment (antoine2711) [reply docs/src/how-tos/contribute_to_or.md:14]: @kushthedude: thank you very much for your corrections!

I'm even before the draft stage with this document, I will for sure need more help.

Regards, A.

File: docs/src/how-tos/build_from_source.md (hunk 1 above)
Comment (antoine2711) [reply docs/src/how-tos/build_from_source.md]: Done.

**Comment by kushthedude** [comment]
> I think we need to discuss whether these should go into this documentation or on a separate platform. Personally I think it would be nice to have it in the same place, to encourage users to turn into contributors, and I don't think it would distract users too much if that is added in a section towards the end of the menu on the left.
> 
> But I think @ostephens prefers to keep this separate, which would also make sense.

@wetneb Do we have any update on this? What I was thinking is to keep our `.md` files in docs folder & we can keep the code related to docusaurus in `website` folder or seperate repo, we can write a bash script to pre-process our docs somewhat similar to [this](https://github.com/thanos-io/thanos/blob/master/scripts/websitepreprocess.sh). Let me know if we have finalised something or discussion is still going on.

**Comment by wetneb** [comment]
We are almost at the end of the technical writer hiring process, thanks for your patience, this should be moving ahead pretty soon!

**Comment by kushthedude** [comment]
> We are almost at the end of the technical writer hiring process, thanks for your patience, this should be moving ahead pretty soon!

Oh, nice!

**Comment by wetneb** [comment]
I have merged the changes introduced by this PR and the new structure of the documentation and will merge this PR soon unless anyone has concerns.
Sorry @antoine2711 for the time it took to merge this!

**Review by wetneb — APPROVED** [review]

**Pull request merged**